torch/_dynamo/utils.py

```diff
@@ -139,6 +139,28 @@ def print_time_report():
     print(out)
 
 
+# NB: There may be multiple TimedFunc with same key
+@dataclasses.dataclass
+class TimedFunc:
+    key: str
+    children: collections.OrderedDict[str, "TimedFunc"] = dataclasses.field(
+        default_factory=collections.OrderedDict
+    )
+    metrics: List[float] = dataclasses.field(default_factory=list)
+
+
+ROOT_TIMED_FUNC = TimedFunc("<root>")
+CURRENT_TIMED_FUNC = ROOT_TIMED_FUNC
+
+
+def render_timed_func(t: TimedFunc, level: int = 0):
+    out = ""
+    out += "  " * level + t.key + "\n"
+    for c in t.children.values():
+        out += render_timed_func(c, level + 1)
```

Comment:
was tempted to say, why don't you just drop the tabulate and render the times on the same lines as you print the call hierarchy?  but, you don't have to do this its still useful

Reply:
I can! So should I tabulate this in that case? So all the times are lined up. But then maybe it would be useful to also print cum/tot??? Should I try to copy paste the python profiler hierarchical output?